tests/Unit/PointwiseFunctions/GeneralRelativity/Test_WeylMagnetic.cpp

```diff
@@ -0,0 +1,114 @@
+  const auto nn_distribution = make_not_null(&distribution);
+
+  *grad_extrinsic_curvature = make_with_random_values<tnsr::ijj<DataType, 3>>(
+      nn_generator, nn_distribution, used_for_size);
+  std::uniform_real_distribution<> metric_distribution(-0.03, 0.03);
+  const auto nn_metric_distribution = make_not_null(&metric_distribution);
+  *spatial_metric = make_with_random_values<tnsr::ii<DataType, 3>>(
+      nn_generator, nn_metric_distribution, used_for_size);
+  for (size_t i = 0; i < 3; ++i) {
+    spatial_metric->get(i, i) += 1.0;
+  }
+}
+
+template <typename DataType>
+void test_compute_item_in_databox(const DataType& used_for_size) noexcept {
+  TestHelpers::db::test_compute_tag<
+      gr::Tags::WeylMagneticCompute<3, Frame::Inertial, DataType>>(
+      "WeylMagnetic");
+
+  auto grad_extrinsic_curvature = make_with_value<tnsr::ijj<DataType, 3>>(
+      used_for_size, std::numeric_limits<double>::signaling_NaN());
+  auto spatial_metric = make_with_value<tnsr::ii<DataType, 3>>(
+      used_for_size, std::numeric_limits<double>::signaling_NaN());
+  make_random_tensors(make_not_null(&grad_extrinsic_curvature),
+                      make_not_null(&spatial_metric), used_for_size);
+
+  const auto box = db::create<
+      db::AddSimpleTags<::Tags::deriv<gr::Tags::ExtrinsicCurvature<
+                                          3, Frame::Inertial, DataType>,
+                                      tmpl::size_t<3>, Frame::Inertial>,
+                        gr::Tags::SpatialMetric<3, Frame::Inertial, DataType>>,
+      db::AddComputeTags<
+          gr::Tags::WeylMagneticCompute<3, Frame::Inertial, DataType>>>(
+      grad_extrinsic_curvature, spatial_metric);
+
+  const auto expected =
+      gr::weyl_magnetic(grad_extrinsic_curvature, spatial_metric);
+  CHECK_ITERABLE_APPROX(
+      (db::get<gr::Tags::WeylMagnetic<3, Frame::Inertial, DataType>>(box)),
+      expected);
+}
+
+template <size_t SpatialDim, typename DataType>
+void test_weyl_magnetic(const DataType& used_for_size) {
+  auto grad_extrinsic_curvature =
+      make_with_value<tnsr::ijj<DataType, SpatialDim>>(
+          used_for_size, std::numeric_limits<double>::signaling_NaN());
+  auto spatial_metric = make_with_value<tnsr::ii<DataType, SpatialDim>>(
+      used_for_size, std::numeric_limits<double>::signaling_NaN());
+  make_random_tensors(make_not_null(&grad_extrinsic_curvature),
+                      make_not_null(&spatial_metric), used_for_size);
+
+  const auto cpp_weyl_magnetic =
+      gr::weyl_magnetic(grad_extrinsic_curvature, spatial_metric);
+  const auto python_weyl_magnetic =
+      pypp::call<tnsr::ii<DataType, SpatialDim, Frame::Inertial>>(
+          "WeylMagnetic", "weyl_magnetic_tensor", grad_extrinsic_curvature,
+          spatial_metric);
+
+  CHECK_ITERABLE_APPROX(cpp_weyl_magnetic, python_weyl_magnetic);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.PointwiseFunctions.GeneralRelativity.WeylMagnetic",
```

Comment:
The functions were defined for arbitrary dimension, and I think these tests only test for spatial dimension of 3 -- you should test the 1 and 2 dimensional cases as well.

Reply:
The function is only defined for 3D, because it uses Levi Civita symbols (it's basically a cross product). Perhaps just remove the template on dimension? Or leave it with a static assert if there's some reason we need the template on dimension instead of hard coding 3D